### PR TITLE
Require user to sign in after accepting invitation

### PR DESCRIPTION
**Pull request by mike29736**
https://trello.com/c/SWVUwQHc/101-consider-requiring-users-to-sign-in-after-accepting-invitation

To mitigate the security risk posed by leaked invitation tokens, we're
no longer automatically signing a user in when they accept an
invitation.

This is a feature of Devise's Invitable module, so not a lot of work on
our part (to enable or disable).

### CODE REVIEW

File: test/integration/inviting_users_test.rb
Comment (floehopper) [inline test/integration/inviting_users_test.rb:6]: Good catch! 😬

File: test/integration/inviting_users_test.rb
Comment (floehopper) [inline test/integration/inviting_users_test.rb:25]: Does this page make it clear that you now need to sign in? Perhaps it's obvious. I mainly want to make sure we're not going to introduce a whole new class of support tickets.

File: config/initializers/devise.rb
Comment (floehopper) [inline config/initializers/devise.rb:123]: Nice!

This makes me wonder 🤔  if we're missing any other configuration options from newer versions of Devise, but that's definitely out-of-scope for this PR!

**Review by floehopper — APPROVED** [review]
This looks good to me - well done on finding the Devise configuration option!

My only slight reservation is about making best efforts to ensure the new UX doesn't cause a rise in support tickets. I haven't actually tried the workflow out myself, so I might well be making a mountain out of a mole hill!

Anyway, I'm happy to leave the above to your discretion, so I'm going to approve the PR.

File: config/initializers/devise.rb
Comment (mike29736) [reply config/initializers/devise.rb:123]: I think it'd be worth regenerating that file just for the improved explanations of some of the config options in the latest version. Anyhow: https://trello.com/c/ARpxcKEA/131-update-signons-devise-and-extensions-config

File: test/integration/inviting_users_test.rb
Comment (mike29736) [reply test/integration/inviting_users_test.rb:25]: I'll ask the team for a second opinion before I merge it.

But here's the page you land on from the invite link:

Screenshot from 2023-08-30 10-09-27

And here's the next page:

Screenshot from 2023-08-30 10-10-03

Comment (floehopper) [reply test/integration/inviting_users_test.rb:25]: Ah, nice! I reckon it's fine, but probably worth a 2nd opinion.

**Pull request merged**
